eclipse-che/che-operator
# Add attributes to set gateway container log level in Che CR

### What does this PR do?
Fixes https://github.com/eclipse/che/issues/22415 by introducing two new fields in the Che CR:
* `spec.networking.auth.gateway.traefikLogLevel`, which takes `DEBUG`, `INFO`, `WARN`, `ERROR`, `FATAL`, or `PANIC`. Defaults to `INFO`
* `spec.networking.auth.gateway.kubeRbacProxyLogLevel`, which takes an integer. Defaults to `0`

### Screenshot/screencast of this PR


<details>
<summary>Verbose logs in the `gateway` container in the `che-gateway` pod</summary>

<img width="996" alt="image" src="https://github.com/eclipse-che/che-operator/assets/83611742/77e10338-ac92-462d-aff3-ddf5ecf3b2d2">

</details>

<details>
<summary>Verbose logs in the `kube-rbac-proxy` container in the `che-gateway` pod</summary>

<img width="998" alt="image" src="https://github.com/eclipse-che/che-operator/assets/83611742/95fd2770-434c-4ec4-a899-198339f71701">

</details>

<details>
<summary>New Che CR fields</summary>

<img width="714" alt="image" src="https://github.com/eclipse-che/che-operator/assets/83611742/8b5a3985-f565-45f6-94c4-4f7114bcc858">

</details>

### What issues does this PR fix or reference?

 https://github.com/eclipse/che/issues/22415 

### How to test this PR?


1. Checkout this PR branch, and run ` ./build/scripts/olm/test-catalog-from-sources.sh` to install Eclipse Che on a cluster
2. After installation, make the `gateway` and `kube-rbac-proxy` container logs more verbose by setting the following fields in the CheCluster CR:
```
spec:
  networking:
    auth:
      gateway:
        kubeRbacProxy:
          logLevel: 10
        traefik:
          logLevel: DEBUG
```
3. Wait 1 minute until the new `che-gateway` pod is deployed
4. Start a workspace and wait until the editor opens
5. Check the `gateway` and `kube-rbac-proxy` container logs within the `che-gateway` pod in the `eclipse-che` namespace
```
$ POD=$(oc get pods -o name -n eclipse-che | grep che-gateway)

# get gateway logs
$ oc logs -c gateway $POD -n eclipse-che

# get kube-rbac-proxy logs
$ oc logs -c kube-rbac-proxy $POD -n eclipse-che
```
6. Verify that the container logs are more verbose.

For the `gateway` logs, check that there are logs with the string `level=debug`
Example logs:
```
time="2023-08-29T19:04:41Z" level=debug msg="Starting TCP Server" entryPointName=http
time="2023-08-29T19:04:41Z" level=debug msg="Starting TCP Server" entryPointName=sink
```

For the `kube-rbac-proxy` logs, check that there are logs with the strings `Request Body:` and `Response Headers:`
Example logs:
```
I0829 19:05:54.039945       1 request.go:1097] Request Body: {"kind":"TokenReview","apiVersion":"authentication.k8s.io/v1","metadata":{"creationTimestamp":null},"spec":{"token":"<REDACTED>"},"status":{"user":{}}}
...

I0829 19:05:54.054710       1 round_trippers.go:443] POST https://172.30.0.1:443/apis/authentication.k8s.io/v1/tokenreviews 201 Created in 14 milliseconds
I0829 19:05:54.054735       1 round_trippers.go:449] Response Headers:
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
